compiler-cli/src/dependencies.rs

```diff
@@ -981,6 +1014,15 @@ impl dependency::PackageFetcher for PackageFetcher {
         &self,
         package: &str,
     ) -> Result<hexpm::Package, Box<dyn std::error::Error>> {
+        {
+            let runtime_cache = self.runtime_cache.borrow();
+            let result = runtime_cache.get(package);
+
+            if let Some(result) = result {
+                return Ok(result.clone());
```

Comment:
Rather than clone the entire package lets record the highest version here.

Reply:
Oh! This function is originally used by the `DependencyProvider`. But as the major version need the same info, but to avoid multiple http calls to hexpm, I've added caching here.

I think the way to avoid cloning would be to have PackageFetcher own package (and return a ref) but that would have to touch the dependency resolution part so I didn't want to include in this PR (increased scope).

Reply:
Here's an implementation with PackageFetcher owning packages and RefCounting: https://github.com/andho/gleam/commit/851b9a6b309e9739f1d2b44e8178831f378f00eb

Reply:
Ah I just realized I never wired up the same `PackageFetcher` to be used in other parts of the dependency resolution so the cache is not even used. I'll see about getting that wired up soon.

Reply:
`PackageFetcher` is now passed by reference in all instances and instantiated in `list` and `download` only.

Reply:
It would be best to do that as to not clone everything. Do you think it would be complex or involved?

Reply:
Changed to pass package as refs using Rc. DependencyProvider still needs to clone the package because it sorts the releases field before passing to pubgrub.